test/e2e/service_test.go

```diff
@@ -194,7 +194,7 @@ func (data *TestData) createAgnhostServiceAndBackendPods(t *testing.T, name, nam
 		ipProtocol = corev1.IPv6Protocol
 	}
 	require.NoError(t, data.podWaitForRunning(defaultTimeout, name, namespace))
-	svc, err := data.CreateService(name, namespace, 80, 80, map[string]string{"app": "agnhost"}, false, false, svcType, &ipProtocol)
+	svc, err := data.CreateService(name, namespace, 80, 80, map[string]string{"app": "agnhost", "antrea-e2e": name}, false, false, svcType, &ipProtocol)
```

Comment:
I doubt if this change really works. Lable `"antrea-e2e":$podName` is also added on all Pods automaticallly which are created by function `createPodOnNode`. If `"app": "agnhost"` is matching other Pods created not in this case, it should be the same.

If the target of this change is to use a different Service selector to get accurate Pods, we should introduce a different label/selector key in the created object from the default settings?

Reply:
> Lable "antrea-e2e":$podName is also added on all Pods automaticallly which are created by function `createPodOnNode`

That's the purpose of the change, podName is different, right? `createPodOnNode(pod1)` and `createPodOnNode(pod2)` will get two Pods with different labels: `{"app": "agnhost", "antrea-e2e": "pod1"}`, `{"app": "agnhost", "antrea-e2e": "pod2"}`, and service1 only selects the former and service2 only selects the latter, anything wrong?

Reply:
@wenyingd does the explanation address your question?

Reply:
Yes, thanks for the explanation.